backend/app/controllers/spree/admin/resource_controller.rb

```diff
@@ -134,7 +134,16 @@ def belongs_to(model_name, options = {})
     end
   end
 
+  def product_exists?
```

Comment:
This method only makes sense if the parent is a product, but the logic could be generalized to handle any kind of parent, right?

Reply:
Thanks for this suggestion, I refactored to generalize the flashes and redirects into the base_controller :+1: